src/App.tsx

```diff
@@ -14,11 +15,96 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[];
+  isShowContent: boolean;
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: goodsFromServer,
```

Comment:
```suggestion
    goods: [...goodsFromServer],
```
then it's no need to make a copy of the array in every method